ibm/mas_devops/roles/entitlement_key_rotation/README.md

```diff
@@ -0,0 +1,117 @@
+- Environment Variable: `SLS_NAMESPACE`
+- Default Value: `ibm-sls`
+
+Role Variables - Advanced mode
+-------------------------------------------------------------------------------
+
+Use the following variables to change the default behavior of this role to only rotate the entitlement key for specific clusters or namespaces, instead of running it for all MAS related namespaces.
+
+### mas_clusters_entitlement_key_rotation_list
+- Optionally define a list of clusters to loop through the entitlement key rotation.
+- Environment Variable: `MAS_CLUSTERS_ENTITLEMENT_KEY_ROTATION_LIST`
+- Default Value: If not set, the `cluster_name` property will be used to target the cluster while executing this role.
+- Example:
+`export MAS_CLUSTERS_ENTITLEMENT_KEY_ROTATION_LIST='cluster1,cluster2'`
+
+### mas_namespaces_entitlement_key_rotation_list
+- Optionally define a specific list of namespaces to loop through the entitlement key rotation.
+- Environment Variable: `MAS_NAMESPACES_ENTITLEMENT_KEY_ROTATION_LIST`
+- Default Value: If not set, all MAS related namespaces for all MAS instances will be target for entitlement key rotation.
+- Example:
+`export MAS_NAMESPACES_ENTITLEMENT_KEY_ROTATION_LIST='ibm-sls,openshift-marketplace'`
+
+Example Playbook
+-------------------------------------------------------------------------------
+
+Rotate entitlement credentials across all MAS instances for a given target cluster:
+
+```yaml
+- hosts: localhost
+  any_errors_fatal: true
+  vars:
+    cluster_name: "{{ lookup('env', 'CLUSTER_NAME') }}"
+    cluster_type: "{{ lookup('env', 'CLUSTER_TYPE') }}"
+    ibmcloud_apikey: "{{ lookup('env', 'IBMCLOUD_APIKEY') }}"
+    artifactory_username: "{{ lookup('env', 'ARTIFACTORY_USERNAME') }}"
+    artifactory_token: "{{ lookup('env', 'ARTIFACTORY_TOKEN') }}"
+    mas_entitlement_username: "{{ lookup('env', 'MAS_ENTITLEMENT_USERNAME') }}"
+    mas_entitlement_key: "{{ lookup('env', 'MAS_ENTITLEMENT_KEY') }}"
+  roles:
+    - ibm.mas_devops.entitlement_key_rotation
+```
+
+Rotate entitlement credentials across a specific list of namespaces, targeting multiple clusters:
+
+```yaml
+- hosts: localhost
+  any_errors_fatal: true
+  vars:
+    cluster_name: "{{ lookup('env', 'CLUSTER_NAME') }}" # this is the original cluster that will keep the login session context at the end of the rotation loop.
+    cluster_type: "{{ lookup('env', 'CLUSTER_TYPE') }}"
+    ibmcloud_apikey: "{{ lookup('env', 'IBMCLOUD_APIKEY') }}"
+    artifactory_username: "{{ lookup('env', 'ARTIFACTORY_USERNAME') }}"
+    artifactory_token: "{{ lookup('env', 'ARTIFACTORY_TOKEN') }}"
+    mas_entitlement_username: "{{ lookup('env', 'MAS_ENTITLEMENT_USERNAME') }}"
+    mas_entitlement_key: "{{ lookup('env', 'MAS_ENTITLEMENT_KEY') }}"
+    mas_clusters_entitlement_key_roration_list: "{{ lookup('env', 'MAS_CLUSTERS_ENTITLEMENT_KEY_ROTATION_LIST') }}"
```

Comment:
typo in "roration" and in the next line below this

Reply:
Fixed